src/ng/directive/ngOptions.js

```diff
@@ -420,12 +420,19 @@ var ngOptionsDirective = ['$compile', '$document', '$parse', function($compile,
 
       // The emptyOption allows the application developer to provide their own custom "empty"
       // option when the viewValue does not match any of the option values.
-      for (var i = 0, children = selectElement.children(), ii = children.length; i < ii; i++) {
-        if (children[i].value === '') {
-          selectCtrl.hasEmptyOption = true;
-          selectCtrl.emptyOption = children.eq(i);
-          break;
-        }
+      var optionsToRemove = [];
+      for (var i = 0, children = selectElement[0].childNodes, ii = children.length; i < ii; i++) {
+          if (!selectCtrl.hasEmptyOption
+              && (children[i].value === '' || children[i].nodeType === NODE_TYPE_COMMENT)) {
```

Comment:
True. I'm doing changes to my commit. I will simplify this whole part and leave it as close to the previous code.